OloEngine/src/OloEngine/Serialization/MeshBinarySerializer.cpp

```
@@ -0,0 +1,1317 @@
+
+            OMeshFormat::SkeletonHeader skelHeader;
+            skelHeader.BoneCount = boneCount;
+            WriteBytes(payload, &skelHeader, sizeof(skelHeader));
+
+            // Parent indices
+            WriteBytes(payload, skeleton->m_ParentIndices.data(), boneCount * sizeof(i32));
+
+            // Transform arrays (7 arrays of mat4)
+            auto const writeMat4Array = [&](const std::vector<glm::mat4>& arr)
+            {
+                if (arr.size() >= boneCount)
```

Clang (14.0.6)

[warning] if with identical then and else branches

(bugprone-branch-clone)